### ISSUE TITLE
Misleading ns.nFormat() magnitude letter

### ISSUE THREAD

**Issue by TheMas3212** [bitburner-official/bitburner-src#62]
[bimsherwood]
Hello,

My currently installed version is 1.6.4 (00628A8C) on Windows installed via Steam.

I have noticed that `ns.nFormat()` sometimes emits a confusing combination of a rounded-up mantissa with the magnitude letter for the unrounded number.

For example:

`1,000,000,000 - 1` formatted with `"0.00 a"` becomes `"1.00 m"`.

The reproduction is as follows:

```
/** @param {NS} ns */
export async function main(ns) {
    ns.tprint(ns.nFormat(1000*1000*1000 - 1, "0.00 a"));
}
```

I noticed I get the same result on the web console on the Numeral.js website. It seems like a massive pain to fix, as Numeral.js looks abandoned.

(Please excuse me if I make a faux-pas, this is the first issue I've posted.)

**Comments From Original Issue**

[Snarling]
> Unfortunately if it's a problem with base numeraljs it will probably be difficult to fix directly in bitburner.
> 
> Searching for up to date numeral alternatives, I found numbro which was forked from numeral and is actively maintained and also available via npm. `numbro(1000*1000*1000 - 1).format("0.00 a")` outputs the expected 1.00 b. So switching over to numbro or another alternative would be a way we can address this bug.

**Comment by OneOfMany07** [comment]
I'm also fighting with nFormat's results.

nFormat(1024 * 1024 * 1024 * 1024, '0.[000]ib') gives '1099511627776' instead of the expected '1TiB'. '0.[000] ib' gives the expected '1 TiB', but '0.[000ib] is needed for '1TiB' output (no space between the number and unit).

I've also been confused by the period needed to be included ('0[.000]a') or excluded ('0.[000]a') from the optional decimal point entries.  I get different behavior for the two format strings, which is why I settled on the format I tried at first at the top.

I'd agree switching to a recently updated replacement makes sense, if numeraljs is not being looked after.

Version: Steam Bitburner v2.1.0 (8f4636cb)
